apps/passport-server/src/services/zuconnectTripshaSyncService.ts

```diff
@@ -121,8 +121,35 @@ export class ZuconnectTripshaSyncService {
     return traced(NAME, "saveData", async (span) => {
       span?.setAttribute("ticket_count", tickets.length);
 
+      const allTickets = await fetchAllZuconnectTickets(this.context.dbPool);
+      // Tickets we just received from Tripsha only have an "external" ID, so
+      // we should use this to check for existing tickets
+      const existingTicketsByExternalId = new Map(
+        allTickets.map((ticket) => [ticket.external_ticket_id, ticket])
+      );
+      // Which tickets are new or updated?
+      const newOrUpdatedTickets = tickets.filter((ticket) => {
+        // If we don't have it in the DB, it's new
+        if (!existingTicketsByExternalId.has(ticket.id)) {
+          return true;
+        } else {
+          // If we do have it but with different values, it's updated
+          const existingTicket = existingTicketsByExternalId.get(ticket.id);
+          if (
+            existingTicket?.attendee_email !== ticket.email ||
+            existingTicket.attendee_name !== ticket.fullName ||
+            existingTicket.product_id !==
+              ZUCONNECT_PRODUCT_ID_MAPPINGS[ticket.ticketName].id
+          ) {
+            return true;
+          }
```

Comment:
worth extracting this to a separate function, kind of how we do it for devconnect tickets.